mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -407,6 +449,99 @@ private[classification] class MultiClassSummarizer extends Serializable {
   }
 }
 
+@Experimental
+class LogisticRegressionTrainingSummary private[classification] (
+    predictions: DataFrame,
+    probabilityCol: String,
+    labelCol: String,
+    val objectiveHistory: Array[Double])
+  extends LogisticRegressionSummary(predictions, probabilityCol, labelCol) {
+
+  /** Number of training iterations until termination */
+  val totalIterations = objectiveHistory.length
+
+}
+
+@Experimental
+class LogisticRegressionSummary private[classification] (
+  @transient val predictions: DataFrame,
+  val probabilityCol: String,
+  val labelCol: String) extends Serializable {
+
+  /** Returns a BinaryClassificationMetrics object.
+  */
+  @transient val metrics = new BinaryClassificationMetrics(
+    predictions.select(probabilityCol, labelCol).map {
+      case Row(score: Vector, label: Double) => (score(1), label)
+    }
+  )
+
+  /**
+   * Returns the receiver operating characteristic (ROC) curve,
+   * which is an Dataframe having two fields (false positive rate, true positive rate)
+   * with (0.0, 0.0) prepended and (1.0, 1.0) appended to it.
+   * Every possible probability obtained in transforming the dataset are used
+   * as thresholds used in calculating the FPR and TPR.
+   */
+  @transient val roc: DataFrame = {
```

Comment:
This should be a `def`; we only want to lazily evaluate `roc` if the user asks for it (same thing is going on in `BinaryClassificationMetrics`). Ditto for others